pyarr/request_api.py

```diff
@@ -99,8 +133,48 @@ def request_del(self, path, params=None, data=None):
             Object: Response object from requests
         """
         headers = {"X-Api-Key": self.api_key}
-        request_url = "{url}{path}".format(url=self.host_url, path=path)
-        res = self.session.delete(
-            request_url, headers=headers, params=params, json=data, auth=self.auth
-        )
-        return res.json()
+        try:
+            res = self.session.delete(
+                self._request_url(path, ver_uri),
+                headers=headers,
+                params=params,
+                json=data,
+                auth=self.auth,
+            )
+        except requests.Timeout as exception:
+            raise PyarrConnectionError(
+                "Timeout occurred while connecting to API"
+            ) from exception
+        return self._check_status_codes(res)
+
+    def _check_status_codes(self, res):
+        """Check the response status code and error or return results
+
+        Args:
+            res (str): JSON or Text response from API Call
+
+        Raises:
+            PyarrUnauthorizedError: Invalid API Key
+            PyarrAccessRestricted: Invalid Permissions
+            PyarrResourceNotFound: Incorrect Resource
+            PyarrBadGateway: Bad Gateway
+
+        Returns:
+            JSON: Array
+        """
+        if res.status_code == 401:
+            raise PyarrUnauthorizedError(
+                "Unauthorized. Please ensure valid API Key is used.", {}
+            )
+        if res.status_code == 403:
+            raise PyarrAccessRestricted(
+                "Access restricted. Please ensure API Key has correct permissions", {}
+            )
+        if res.status_code == 404:
+            raise PyarrResourceNotFound("Resource not found")
+        if res.status_code == 502:
+            raise PyarrBadGateway("Bad Gateway. Check your server is accessible")
+        content_type = res.headers.get("Content-Type", "")
+        if "application/json" in content_type:
+            return res.json()
```

Comment:
Having this here makes the function name a misnomer.

Consider moving it out, or renaming this function to something like `_process_response`

Reply:
agreed, updated naming